src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
```

Comment:
FYI: Default value for coords would work only if consstructor was called with only two first params (name and weight), also chipVersion param wouldn't have any default value. I suggest removing default value cause we always call this constructor with atleast blank object for coords.
```suggestion
  constructor(name, weight, coords, chipVersion) {
```

Reply:
I removed default value, thank you for advice.